packages/workspace/src/core/nx-deps/nx-deps-cache.ts

```diff
@@ -74,68 +79,125 @@ export function readCache(): false | ProjectGraphCache {
 }
 
 export function writeCache(
-  rootFiles: FileData[],
+  packageJsonDeps: Record<string, string>,
+  nxJson: NxJsonConfiguration,
+  tsConfig: { compilerOptions: { paths: { [k: string]: any } } },
   projectGraph: ProjectGraph
 ): void {
   performance.mark('write cache:start');
-  writeJsonFile(nxDepsPath, {
-    version: '2.0',
-    rootFiles,
+  const nxJsonPlugins = (nxJson.plugins || []).map((p) => ({
+    name: p,
+    version: packageJsonDeps[p],
+  }));
+  const newValue: ProjectGraphCache = {
+    version: '3.0',
+    deps: packageJsonDeps,
+    pathMappings: tsConfig.compilerOptions.paths,
+    nxJsonPlugins,
     nodes: projectGraph.nodes,
     dependencies: projectGraph.dependencies,
-  });
+  };
+  writeJsonFile(nxDepsPath, newValue);
   performance.mark('write cache:end');
   performance.measure('write cache', 'write cache:start', 'write cache:end');
 }
 
-export function differentFromCache(
+export function shouldRecomputeWholeGraph(
```

Comment:
What happens if `implicitDependencies` in `nx.json` changes?

Reply:
Handled. We remove those projects from the file map now.